### PR TITLE
Snow 1105625 git list branches tags

### PR DESCRIPTION
**Pull request by sfc-gh-pczajka**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Add `git list-branches` and `git list-tags` commands

### CODE REVIEW

File: src/snowflake/cli/plugins/git/manager.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from snowflake.cli.api.sql_execution import SqlExecutionMixin
+
+
+class GitManager(SqlExecutionMixin):
+    def show(self, object_type: str, repo_name: str):
+        assert object_type in ["branches", "tags"]
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/git/manager.py:6]: Personally I would separate this into two methods. If not, then let's introduce enum for branches and tags

File: src/snowflake/cli/plugins/git/commands.py [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
+import typer
 from snowflake.cli.api.commands.snow_typer import SnowTyper
-from snowflake.cli.api.output.types import (
-    CommandResult,
-    MessageResult,
-)
+from snowflake.cli.api.output.types import CommandResult, QueryResult
+from snowflake.cli.plugins.git.manager import GitManager
 
 app = SnowTyper(
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/git/commands.py:8]: Let's make this a hidden group

Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/git/commands.py:8]: Done!

File: src/snowflake/cli/plugins/git/manager.py (hunk 1 above)
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/git/manager.py]: I've separated the commands - An enum would be an overkill for now.

File: src/snowflake/cli/plugins/git/commands.py [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
     name="git",
     help="Manages git repositories in Snowflake.",
 )
 log = logging.getLogger(__name__)
 
+RepoNameArgument = typer.Argument(help="name of the git repository")
```
[the remark this answers is not in the document]
Comment (sfc-gh-pczajka) [reply src/snowflake/cli/plugins/git/commands.py]: done!

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**